.github/workflows/test.yml

```
@@ -15,24 +15,90 @@ on:
+          pip install -r requirements.txt
+          pip install -r requirements-dev.txt
+
+      - name: Run Ruff Linter (Auto-Fix Imports)
+        run: ruff check --fix --select I --unsafe-fixes --output-format=github .
+
+      - name: Run Ruff Formatter (Auto-Fix)
+        run: ruff format .
+
+      - name: Commit Ruff Fixes
+        if: always()
+        run: |
```

actionlint (1.7.8)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)